new_app_setup.py

```diff
@@ -0,0 +1,266 @@
+
+    namespace_filename = team_name+"_namespace.yml"
+
+    pods_filename = team_name+"_pods.yml"
+
+    create_kubernetes_namespace_yml(namespace_filename, team_name)
+
+    create_kubernetes(namespace_filename, team_name, "namespace")
+
+    if args.RAM is not None:
+        ram_requested = args.RAM
+    else:
+        ram_requested = '2G'  # default
+
+    if args.CPU is not None:
+        cpu_requested = args.CPU
+    else:
+        cpu_requested = '1'  # default
+
+    create_kubernetes_pods_yml(pods_filename, team_name, ram_requested,
+                               cpu_requested)
+
+    create_kubernetes(pods_filename, team_name, "pods")
+
+    config_file = get_congif_yml_file(team_name)
+
+    path_to_congif_file = os.path.join(base_path, config_file)
+
+    send_config_email_to_users(team_name, args.group_user, config_file)
+
+    if args.list_of_users is not None:
+        send_config_email_to_users(team_name, args.list_of_users, config_file)
+
+
+def create_kubernetes_namespace_yml(namespace_filename, team_name):
+
+    tag = "Hack_For_A_Cause_2019"
+
+    yml_file_kubernetes_data = dict(
+
+        kind='Namespace',
+        apiVersion='v1',
+        metadata=dict(
+            name=team_name,
+            labels=dict(
+                name=tag
+                )
+            )
+        )
+
+    with open(namespace_filename, 'w') as outfile:
+        yaml.dump(yml_file_kubernetes_data, outfile, default_flow_style=False)
+
+    print("The YAML file for {} was created and was "
+          "saved as {}".format(team_name, namespace_filename))
+
+    return yml_file_kubernetes_data
+
+
+def create_kubernetes_pods_yml(pods_filename, team_name, ram_requested,
+                               cpu_requested):
+    yml_file_kubernetes_data = dict(
+        kind='Pod',
+        apiVersion='v1',
+        metadata=dict(
+            name=team_name,
+            namespace=team_name,
+            ),
+        spec=dict(
+                containers=dict(
+                    name=team_name,  # QUESTION HERE -> see below
+                    resources=dict(
+                        limits=dict(
+                            cpu="1",
+                            memory="2G"
+                        ),
+                        requests=dict(
+                            cpu=cpu_requested,
+                            memory=ram_requested,
+                        )
+                    )
+                )
+            )
+        )
+# QUESTION: When running, I am getting the warning:
+
+# "error: error validating "test-team-name_pods.yml": error validating
+# data: ValidationError(Pod.spec.containers): invalid type for
+# io.k8s.api.core.v1.PodSpec.containers: got "map", expected "array";
+# if you choose to ignore these errors, turn validation off with
+# --validate=false
+# and I know that is because name does not have the hyphen indicating an array
+# in front of it, but I cannot seem to figure out how to do this...
+# anything I'm missing?
```

Comment:
it's saying "containers: got "map", expected "array" - that is containers on line 117 is supposed to be a **list** of containers but you've given it a dict. A dict in python == a map in YAML. So line 117 should be something like:

```
spec = dict(
    containers=[
        dict( name=team_name, ....)
```

Note that it's much more common in Python to just use `{}` for a dict instead of the explicit dict constructor. That is, `dict(a=1, b=2)` and `{'a': 1, 'b': 2}` create exactly the same object by the 2nd is a much more common way to see it written.